crates/datasources/src/mysql/mod.rs

```diff
@@ -372,6 +374,62 @@ impl TableProvider for MysqlTableProvider {
             query,
             arrow_schema: projected_schema,
             metrics: ExecutionPlanMetricsSet::new(),
+            is_read_only: true,
+        }))
+    }
+
+    async fn insert_into(
+        &self,
+        state: &SessionState,
+        input: Arc<dyn ExecutionPlan>,
+        _overwrite: bool,
+    ) -> DatafusionResult<Arc<dyn ExecutionPlan>> {
+        let mut values = String::new();
+
+        let mut input = execute_stream(input, state.task_ctx())?;
+        while let Some(batch) = input.next().await {
+            let batch = batch?;
+
+            let mut row_sep = "";
+            for row_idx in 0..batch.num_rows() {
+                write!(&mut values, "{row_sep}(")?;
+                row_sep = ",\n\t";
+
+                let mut col_sep = "";
+                for col in batch.columns() {
+                    write!(&mut values, "{col_sep}")?;
+                    col_sep = ", ";
+
+                    let val = ScalarValue::try_from_array(col.as_ref(), row_idx)?;
+                    util::encode_literal_to_text(util::Datasource::MySql, &mut values, &val)
+                        .map_err(|e| DataFusionError::External(Box::new(e)))?;
+                }
+
+                write!(&mut values, ")")?;
+            }
+        }
+
+        if values.is_empty() {
+            let batch = create_count_record_batch(0);
+            let schema = batch.schema();
+            return Ok(Arc::new(MemoryExec::try_new(&[vec![batch]], schema, None)?));
+        }
+
+        let query = format!(
+            "INSERT INTO {}.{} VALUES\n\t{}",
+            self.table_access.schema, self.table_access.name, values
+        );
+
+        debug!(%query, "inserting into mysql datasource");
+
+        Ok(Arc::new(MysqlExec {
+            predicate: "".to_string(),
+            table_access: self.table_access.clone(),
+            accessor: self.accessor.clone(),
+            query,
+            arrow_schema: self.arrow_schema.clone(),
```

Comment:
This seems weird. Shouldn't this be `COUNT_SCHEMA`? Or am I missing something

Reply:
ah yes! checking

Reply:
fixed